File: Cpp/021.cpp
#include <cstdio>
#include <iostream>
using namespace std;

int div_sum(int a)
{
    int sum=0;
    for(int i=1;i<=a/2;i++)
    {
        if(a%i==0) sum+=i;
    }
    return sum;
}

int main()
{
    int sum=0,x;
    for(int i=1;i<=10000;i++)
    {
        x=div_sum(i);
        if(i<x && i==div_sum(x)) sum+=i+x;
    }
    cout<<sum;
}

File: Ruby/021.rb
def divisor_sum(n)
  sum = 1
  i = 2
  while i <= n / 2 do
    if n % i == 0
      sum += i
    end
    i += 1
  end
  sum
end

n = 2
sum = 0

while n <= 10000 do
  x = divisor_sum(n)
  sum += n + x if n < x && n == divisor_sum(x)
  n += 1
end

puts sum